### PR TITLE
OS-xxxx LX procfs users can cause fifofs vs. proc deadlocks

### PR DESCRIPTION
**Pull request by danmcd**
Pinging @citrus-it for extra eyes from OmniOS.

### CODE REVIEW

**Review by tsoome — COMMENTED** [review]
tbh, this function only does something when bp != NULL, so lets just insert at the beginnning:

if (bp == NULL)
    return (0);

and we can move the block comments to it. Ok, well, we actually do just one thing -- we do validate rvp->v_type and return -1 in case of unknown value (do we need to do this when bp == NULL?).

**Review by tsoome — DISMISSED** [review]
Ok, we need to test the type even with bp == NULL.

**Comment by danmcd** [comment]
Inducing the panic-case is still quite difficult.

**Review by bahamat — APPROVED** [review]

**Review by tsoome — DISMISSED** [review]

**Review by citrus-it — APPROVED** [review]
Thanks for tracking this down. I wonder if there should also be a precautionary comment in lxpr_readlink to note that this depends on the behaviour there.

**Comment by danmcd** [comment]
I still have not been able to induce this deadlock.  I'm updating with a copyright and @citrus-it 's suggestion about lxpr_readlink.

**Comment by danmcd** [comment]
I'm putting Andy's comment not in lxpr_readlink, but in lxpr_lookup_fdnode().

File: usr/src/uts/common/brand/lx/procfs/lx_prsubr.c [hunk 1]
```diff
@@ -803,7 +804,11 @@ lxpr_lookup_fdnode(vnode_t *dvp, const char *name)
 	 * bogus-named symlink.  If that's the case, report the type as
 	 * VNON to bypass link-following elsewhere in the vfs system.
 	 *
-	 * See lxpr_readlink for more details.
+	 * See lxpr_readlink for details about certain entries.
+	 *
+	 * Note that lxpr_readlink_fdnode() is being used here as a glorified
+	 * checker of lxfp here, and it will not acquire any further locks given
```
Comment (bahamat) [inline usr/src/uts/common/brand/lx/procfs/lx_prsubr.c:810]: > is being used here as a glorified checker of lxfp here

Could lose one of those "here"s.

**Review by bahamat — APPROVED** [review]

**Comment by danmcd** [comment]
Given the only other things that changed were comments, I'm adding other reviewers to this commit.
